python/ray/data/datasource/datasink.py

```diff
@@ -103,7 +103,7 @@ def supports_distributed_writes(self) -> bool:
         return True
 
     @property
-    def num_rows_per_write(self) -> Optional[int]:
+    def min_rows_per_write(self) -> Optional[int]:
         """The target number of rows to pass to each :meth:`~ray.data.Datasink.write` call.
 
         If ``None``, Ray Data passes a system-chosen number of rows.
```

Comment:
`Datasink` is a developer API. I doubt anyone is overriding this method, but there is a chance this could break third-party implementations

Reply:
Makes sense... but do keep in mind:

```
Developer APIs are lower-level methods explicitly exposed to advanced Ray users and library developers. Their interfaces may change across minor Ray releases.
```

Reply:
Actually, this'll probably break Lance's implementation again (we just broke it recently https://github.com/ray-project/ray/pull/47942#issuecomment-2496067681): https://github.com/lancedb/lance/blob/9c08d2d1ca2ff72bd1c2ea4a84400a7b655b2172/python/python/lance/ray/sink.py#L234-L236.

Maybe let's maintain backwards compabaility? cc @raulchen

Reply:
hmm, the semantics for their parameter there is literally the opposite of ours (ours is min, theirs is max)

Aside from that -- perhaps we can keep this method interface around and log a warning for deprecation?

Reply:
Actually, now I think about it -- this doesn't actually break the implementation right? * In terms of behavior, their expected behavior was different than what we provided
* In terms of actual code correctness, the new state will be as if they didn't implement min_rows_per_write, which will then return None.

So they'll basically be overwriting a function that doesn't get used, but won't break stuff.

Reply:
Makes sense.

I trust your judgement on how we handle (or don't handle) the API change